calculator.js

```diff
@@ -1,5 +1,119 @@
+
+// Refactor to Accept User Input
+
+const schema = {
+  properties:{
+    'num1':{
+      type:'number',
+      message:"num1 must be a number",
+      required: true
+
+    },
+    'num2':{
+      type:'number',
+      message:"num2 must be a number"
+    },
+    'operation':{
+
+    }
+  }
+}
+
+
+const prompt = require('prompt');
+
+const calculatorUserInput = function(error,promptInput){
+  console.log('num1 :' + promptInput.num1);
+  console.log('num2 :' + promptInput.num2);
+  console.log('operation:' + promptInput.operation);
+  console.log(promptInput.num1 + " "+promptInput.operation+" "+promptInput.num2+" = "+ js_calculator(promptInput));
+}
+
+
+
+console.log("WELCOME! Please enter num1, num2, and operations\nExample input:\n========== \nadd,+\nsubtract,-\nmultiply,*\ndivide,/\n========== ");
+prompt.start();
+prompt.get(schema, calculatorUserInput);
+
+
+//js_calculator function
+
+const js_calculator = function(input) {
+
+  let valid_operations = ["add","subtract","multiply","divide","+","-","*","/"];
+
+  if (typeof input.num1 !== "number" || typeof input.num2 !=="number"){
+    return `Invalid input of num1:${input.num1} or num2:${input.num2}`;
+  }
+  if (valid_operations.includes(input.operation) === false ){
+    return `Invalid input of operation:${input.operation}`;
+  }
+
+  if (input.operation === "add" || input.operation === "+"){
+    return input.num1 + input.num2 ;
+  }
+  else if (input.operation === "subtract" || input.operation === "-"){
+    return input.num1 - input.num2 ;
+  }
+  else if (input.operation === "multiply" || input.operation === "*"){
+    return input.num1 * input.num2 ;
+  }
+  else if (input.operation === "divide" || input.operation === "/"){
+    return input.num1 / input.num2 ;
+  }
+  else {
+    return "Error";
+  }
+
+}
```

Comment:
It's funky but ';' belong at the end of function and object definitions as well.
JS is super flexible so it works hard to add them for you when it's pretty sure you forgot one, but our best practice at Ada is to try to add them everywhere they go rather than relying on JS to do that for us.